lib/tapioca/compilers/dsl_compiler.rb

```diff
@@ -22,19 +22,31 @@ class DslCompiler
           requested_constants: T::Array[Module],
           requested_generators: T::Array[T.class_of(Dsl::Base)],
           excluded_generators: T::Array[T.class_of(Dsl::Base)],
-          error_handler: T.nilable(T.proc.params(error: String).void)
+          error_handler: T.nilable(T.proc.params(error: String).void),
+          number_of_workers: T.nilable(Integer),
         ).void
       end
-      def initialize(requested_constants:, requested_generators: [], excluded_generators: [], error_handler: nil)
+      def initialize(
+        requested_constants:,
+        requested_generators: [],
+        excluded_generators: [],
+        error_handler: nil,
+        number_of_workers: nil
+      )
         @generators = T.let(
           gather_generators(requested_generators, excluded_generators),
           T::Enumerable[Dsl::Base]
         )
         @requested_constants = requested_constants
         @error_handler = T.let(error_handler || $stderr.method(:puts), T.proc.params(error: String).void)
+        @number_of_workers = number_of_workers
       end
 
-      sig { params(blk: T.proc.params(constant: Module, rbi: RBI::File).void).void }
+      sig do
+        type_parameters(:T).params(
+          blk: T.proc.params(constant: Module, rbi: RBI::File).returns(T.nilable(T.type_parameter(:T)))
```

Comment:
I don't think the block passed to this methods really returns anything that is nilable. It is just the block we pass to `Executor` inside this method returns either this type parameter or `nil`.
```suggestion
          blk: T.proc.params(constant: Module, rbi: RBI::File).returns(T.type_parameter(:T))
```